### PR TITLE
Suggest re-integration of forked copy that contains many additional projects.

### PR DESCRIPTION
**Pull request by meitar**
Shortly after January 2018, I forked this project and independently began maintaining my own list because, at the time, project maintainers refused to list either non-"Web" software (even though it could be self-hosted) nor the client applications of software that *was* added to this list but that did not, itself, run in a Web browser. [I argued against that approach](https://github.com/Kickball/awesome-selfhosted/pull/1287) and am glad to see the maintainers seem to have changed their minds at least somewhat.

Given the change of heart, I'm opening this pull request to suggest re-integration with my own forked copy. I have been tracking your base copy since January 2018 so there are not too many differences. There are, however, numerous additional self-hostable network services and Web apps in my list that are missing from the original list, as well as minor changes in the wording of several line items in order to more closely conform with the official Awesome List style guide (primarily the removal of prepositions such as "a" and "the" at the beginning of line item descriptions).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -938,6 +966,7 @@ See also <https://en.wikipedia.org/wiki/List_of_streaming_media_systems>, <https
 - [asciiflow](http://asciiflow.com/) - Flow Diagram Drawing Tool. ([Source Code](https://github.com/lewish/asciiflow2)) `GPL-3.0` `Java/JavaScript`
 - [blynk](http://www.blynk.cc/) - Platform with iOS and Android apps to control Arduino, ESP8266, Raspberry Pi and similar microcontroller boards over the Internet. ([Source Code](https://github.com/blynkkk/blynk-server)) `AGPL-3.0` `Java`
 - [CUPS](https://www.cups.org/) - The Common Unix Print System uses Internet Printing Protocol (IPP) to support printing to local and network printers. ([Source Code](https://github.com/apple/cups)) `GPL-2.0` `C`
+- [CyberChef](https://github.com/gchq/CyberChef) - Perform all manner of "cyber" operations within a web browser such as AES, DES and Blowfish encryption and decryption, creating binary and hexdumps, compression and decompression of data, calculating hashes and checksums, IPv6 and X.509 parsing, changing character encodings, and much more. ([Demo](https://gchq.github.io/CyberChef)) `Apahce-2.0` `JavaScript`
```
Comment (evaryont) [inline README.md]: Typo in the license here

File: README.md [hunk 2]
```diff
@@ -948,21 +977,24 @@ See also <https://en.wikipedia.org/wiki/List_of_streaming_media_systems>, <https
...
+- [oTranscribe](http://otranscribe.com/) - Free web app to take the pain out of transcribing recorded interviews. ([Source Code](https://github.com/oTranscribe/oTranscribe)) `MIT` `JavaScript`
 - [PassCheck](https://passcheck.xyz/) - A web application featuring some handy password tools, including a password generator, strength checker and HaveIBeenPwned breach checker. ([Source Code](https://github.com/apacketofsweets/PassCheck)) `MIT` `JavaScript`
-- [ReleaseBell](https://releasebell.com/) - Send release notifications for starred Github repos. ([Source Code](https://git.cloudron.io/cloudron/releasebell)) `MIT` `Nodejs` 
 - [revealjs](https://revealjs.com) - Framework for easily creating beautiful presentations using HTML. ([Demo](https://revealjs.com/), [Source Code](https://github.com/hakimel/reveal.js)) `MIT` `JavaScript`
+- [Revive Adserver](https://www.revive-adserver.com/) - World's most popular free, open source ad serving system. Formerly "OpenX Adserver" and "phpAdsNew." ([Source Code](https://github.com/revive-adserver/revive-adserver)) `GPL-2.0-or-later` `PHP`
```
Comment (evaryont) [inline README.md]: The regex checks that Travis runs checks for a trailing dot, but doesn't know English grammar. Safe to ignore IMO, or is there a way to rephrase it?

**Comment by nodiscc** [comment]
Hi @meitar thanks for staying in touch.

The list is not focused on web stuff but on server-side software. We added support for a "clients" link next to Demo and Source Code link (like `[Clients](http://software.example/clients)`), so I think `M.A.L.P` should be discarded, and a [Clients](https://www.musicpd.org/clients/) link should be added for mpd (it's listed there).

I will check other changes now but at first sight it looks good, thanks

File: README.md [hunk 3]
```diff
@@ -131,6 +132,7 @@ _Business Intelligence_
 _Social Media_
 
 - [IG Monitoring](https://igmonitoring.com) -  `⚠` Instagram Analytics and Stats. ([Demo](https://demo.igmonitoring.com), [Source Code](https://github.com/jakim/ig-monitoring)) `MIT` `PHP`
+- [Socioboard](https://socioboard.org/) - Social media management, analytics, and reporting platform supporting nine social media networks out-of-the-box. ([Source Code](https://github.com/socioboard/Socioboard-Core-3.0)) `GPL-3.0` `C#/JavaScript`
```
Comment (nodiscc) [inline README.md]: Only supports non-FOSS social networks so the `⚠` symbol must be added

File: README.md [hunk 4]
```diff
@@ -140,8 +142,8 @@ Some [Content Management System](#content-management-systems-cms) solutions also
...
-- [ArchiveBox](https://archivebox.io/) - Self-hosted "wayback machine" that creates HTML & screenshot archives of sites from your bookmarks, browsing history, RSS feeds, or other sources. ([Demo](https://archive.sweeting.me), [Source Code](https://github.com/pirate/bookmark-archiver)) `MIT` `Python`
 - [ArchivesSpace](https://archivesspace.org/) - Archives information management application for managing and providing Web access to archives, manuscripts and digital objects. ([Demo](https://archivesspace.org/application/demo/), [Source Code](https://github.com/archivesspace/archivesspace)) `ECL-2.0` `Ruby`
+- [ArchiveBox](https://archivebox.io/) - Self-hosted "wayback machine" that creates HTML & screenshot archives of sites from your bookmarks, browsing history, RSS feeds, or other sources. ([Demo](https://archive.sweeting.me), [Source Code](https://github.com/pirate/bookmark-archiver)) `MIT` `Python`
```
Comment (nodiscc) [inline README.md]: The previous alphabetical ordering was correct

File: README.md [hunk 5]
```diff
@@ -441,6 +445,7 @@ _[SIP](https://en.wikipedia.org/wiki/Session_Initiation_Protocol)/[IPBX](https:/
 - [Orange Forum](http://www.goodoldweb.com/) - Orange Forum is an easy to deploy forum that has minimal dependencies and uses very little javascript. ([Demo](https://groups.goodoldweb.com/), [Source Code](https://github.com/s-gv/orangeforum)) `BSD-3-Clause` `Go`
 - [OSSN](https://www.opensource-socialnetwork.org/) - Open Source Social Network (OSSN) is a social networking software written in PHP. It allows you to make a social networking website and helps your members build social relationships, with people who share similar professional or personal interests. ([Source Code](https://github.com/opensource-socialnetwork/opensource-socialnetwork)) `GPL-2.0` `PHP`
 - [Oxwall](http://www.oxwall.org/) - Oxwall is used for a wide range of projects starting from family sites and custom social networks to collaboration tools and enterprise community solutions. ([Source Code](https://bitbucket.org/oxwall/public)) `CPAL-1.0` `PHP`
+- [Patchwork](https://github.com/ssbc/patchwork) - Decentralized messaging and sharing app built on top of Secure Scuttlebutt (SSB). `AGPL-3.0-only` `JavaScript`
```
Comment (nodiscc) [inline README.md]: Would `Nodejs` platform be more [appropriate](https://github.com/ssbc/patchwork#installation)?

File: README.md [hunk 6]
```diff
@@ -851,16 +866,27 @@ _Games, game servers and control panels._
...
 - [Nominatim](https://nominatim.org/) - Server application for reverse geocoding (address -> coordinates) on OpenStreetMap data. ([Source Code](https://github.com/openstreetmap/Nominatim)) `GPL-2.0` `C`
 - [OpenGTS](http://www.opengts.org/) - Entry-level fleet tracking system. Supports variety of tracking devices and protocols. Comes with rich web-interface and reporting features. ([Demo](http://track.opengts.org/track/Track), [Source Code](https://sourceforge.net/projects/opengts/files/server-base/)) `Apache-2.0` `Java`
-- [OpenStreetMap](http://www.openstreetmap.org/) - OpenStreetMap is a map of the world, created by people like you and free to use under an open license. ([Source Code](https://github.com/openstreetmap/openstreetmap-website)) `GPL-2.0` `Ruby`
+- [OpenStreetMap](https://www.openstreetmap.org/) - Collaborative project to create a free editable map of the world. ([Source Code](https://github.com/openstreetmap/openstreetmap-website)) `GPL-2.0` `Ruby`
```
Comment (nodiscc) [inline README.md]: I like and use Osmand (it's really great) and StreetComplete. However I think we should link to [Clients](https://wiki.openstreetmap.org/wiki/Software) instead, and improve the OSM wiki if needed

**Comment by meitar** [comment]
@nodiscc I believe commit a5516f765dc32b0a435d5320b3844554c6ac64c5 addresses all your points. Thanks for the prompt review.

**Comment by meitar** [comment]
@nodiscc Please have a look at commit 4a5f0d63f7926a4a960c43761534962a8ebe7d5e which should address your latest comments. Thanks.

File: README.md [hunk 7]
```diff
@@ -888,11 +911,12 @@ See also <https://en.wikipedia.org/wiki/List_of_streaming_media_systems>, <https
 - [Airsonic](https://airsonic.github.io/) - Open-source web-based media streamer and jukebox. A fork of Subsonic's last open-source release, before it switched licenses. ([Source Code](https://github.com/airsonic/airsonic), [Clients](https://airsonic.github.io/docs/apps/)) `GPL-3.0` `Java`
 - [AzuraCast](https://www.azuracast.com/) - A modern and accessible self-hosted web radio management suite. ([Source Code](https://github.com/AzuraCast/AzuraCast)) `Apache-2.0` `PHP`
 - [Beets](http://beets.io/) - Music library manager and MusicBrainz tagger (command-line and Web interface). ([Source Code](https://github.com/beetbox/beets)) `MIT` `Python`
+- [Black Candy](https://github.com/aidewoode/black_candy) - Music streaming server built with Rails and Stimulus. ([Demo](http://178.62.101.174/)) `MIT` `Ruby`
```
Comment (nodiscc) [inline README.md]: The demo site is down for me (timeout)

File: README.md [hunk 8]
```diff
@@ -948,21 +973,24 @@ See also <https://en.wikipedia.org/wiki/List_of_streaming_media_systems>, <https
 - [ytdl-webserver](https://github.com/Algram/ytdl-webserver) - Docker-ready webserver for downloading youtube videos. `MIT` `Nodejs`
 - [Kimai](http://www.kimai.org/) - Simple time and project tracking. ([Demo](http://www.kimai.org/demo/), [Source Code](https://github.com/kimai/kimai)) `GPL-3.0` `PHP`
 - [King Phisher](https://github.com/securestate/king-phisher) - King Phisher is a tool for testing and promoting user awareness by simulating real world phishing attacks. `BSD-3-Clause` `Python`
+- [LibreNews](https://librenews.io/) - Decentralized and secure breaking news notification system. ([Source Code](https://github.com/milesmcc/LibreNews-Server/)) `GPL-3.0` `Python`
```
Comment (nodiscc) [inline README.md]: Wouldn't the https://github.com/Kickball/awesome-selfhosted#custom-communication-systems section be a better fit?

File: README.md (hunk 7 above)
Comment (evaryont) [reply README.md]: It's working for me. Maybe the host/process was being upgraded?

File: README.md (hunk 8 above)
Comment (meitar) [reply README.md]: @nodiscc Okay, I moved LibreNews to the section you suggested.

File: README.md [hunk 9]
```diff
@@ -938,6 +963,7 @@ See also <https://en.wikipedia.org/wiki/List_of_streaming_media_systems>, <https
 - [asciiflow](http://asciiflow.com/) - Flow Diagram Drawing Tool. ([Source Code](https://github.com/lewish/asciiflow2)) `GPL-3.0` `Java/JavaScript`
 - [blynk](http://www.blynk.cc/) - Platform with iOS and Android apps to control Arduino, ESP8266, Raspberry Pi and similar microcontroller boards over the Internet. ([Source Code](https://github.com/blynkkk/blynk-server)) `AGPL-3.0` `Java`
 - [CUPS](https://www.cups.org/) - The Common Unix Print System uses Internet Printing Protocol (IPP) to support printing to local and network printers. ([Source Code](https://github.com/apple/cups)) `GPL-2.0` `C`
+- [CyberChef](https://github.com/gchq/CyberChef) - Perform all manner of "cyber" operations within a web browser such as AES, DES and Blowfish encryption and decryption, creating binary and hexdumps, compression and decompression of data, calculating hashes and checksums, IPv6 and X.509 parsing, changing character encodings, and much more. ([Demo](https://gchq.github.io/CyberChef)) `Apache-2.0` `JavaScript`
```
Comment (n8225) [inline README.md]: Please shorten the description to 250 or fewer characters.

File: README.md [hunk 10]
```diff
@@ -1139,6 +1167,7 @@ See also _[Monitoring](#monitoring)_
 - [Kyélà](http://kyela.net/) - Participation polls for group events. ([Demo](https://kyela.net/concert/), [Source Code](https://github.com/abienvenu/Kyela)) `AGPL-3.0` `PHP`
 - [LimeSurvey](https://www.limesurvey.org) - Feature-rich Open Source web based polling software. Supports extensive survey logic. ([Demo](https://demo.limesurvey.org), [Source code](https://github.com/LimeSurvey/LimeSurvey)) `GPL-2.0` `PHP`
 - [Nuages](https://nuages.domainepublic.net/) - Collaborative meeting poll system, similar to doodle or rdvz. ([Source Code](http://git.domainepublic.net/?p=nuages.git;a=tree)) `GPL-3.0` `Python`
+- [Open Event Server](https://github.com/fossasia/open-event-server) - Enables organizers to manage events from concerts to conferences and meet-ups. ([Demo](http://open-event-api.herokuapp.com/)) `GPL-3.0` `Python`
```
Comment (n8225) [inline README.md]: Update demo to https://open-event-api.herokuapp.com/

File: README.md [hunk 11]
```diff
@@ -380,9 +383,10 @@ _Mailing lists servers and mass mailing software - one message to many recipient
 _[IRC](https://en.wikipedia.org/wiki/Internet_Relay_Chat) communication software_
 
 - [Convos](http://convos.by/) - Always online web IRC client. ([Demo](http://demo.convos.by), [Source Code](http://github.com/nordaaker/convos)) `Artistic-2.0` `Perl`
-- [Dispatch](https://github.com/khlieng/dispatch) - A self-hosted web IRC client written in Go. ([Demo](https://dispatch.khlieng.com/connect)) `MIT` `Go`
-- [Kiwi IRC](https://kiwiirc.com/) - A responsive web IRC client with theming support. ([Demo](https://kiwiirc.com/nextclient/)), ([Source Code](https://github.com/kiwiirc/kiwiirc)) `Apache-2.0` `Nodejs`
-- [The Lounge](https://thelounge.github.io/) - A self-hosted web IRC client. ([Demo](https://demo.thelounge.chat/), [Source Code](https://github.com/thelounge/lounge)) `MIT` `Nodejs`
+- [InspIRCd](https://www.inspircd.org/) - Modular IRC server written in C++ for Linux, BSD, Windows, and macOS. ([Source Code](https://github.com/inspircd)) `GFDL-1.2-only` `C++`
```
Comment (n8225) [inline README.md]: Update source code to https://github.com/inspircd/inspircd

File: README.md [hunk 12]
```diff
@@ -851,16 +869,22 @@ _Games, game servers and control panels._
 
 **[`^        back to top        ^`](#)**
 
-- [Graphhopper](https://graphhopper.com/) - Fast routing library and server using OpenStreetMap. ([Source Code](https://github.com/graphhopper/graphhopper)) `Apache-2.0` `Java`
+See also [awesome-gis](https://github.com/sshuair/awesome-gis).
+
+- [GraphHopper](https://graphhopper.com/) - Fast routing library and server using OpenStreetMap. ([Source Code](https://github.com/graphhopper/graphhopper)) `Apache-2.0` `Java`
 - [MapBBCodeShare](https://github.com/MapBBCode/share.mapbbcode.org) - Tool for sharing custom OSM maps. Support for annotated markers, polygons, lines, multi-format import/export, multiple layers, shortlinks. ([Demo](http://share.mapbbcode.org/)) `WTFPL/Other` `PHP`
+- [Open Source Routing Machine (OSRM)](http://project-osrm.org/) - High performance routing engine designed to run on OpenStreetMap data and offering an HTTP API, C++ library interface, and NodeJS wrapper. ([Demo](https://map.project-osrm.org/), [Source Code](https://github.com/Project-OSRM)) `BSD-2-Clause` `C++`
```
Comment (n8225) [inline README.md]: Update source code to https://github.com/Project-OSRM/osrm-backend

**Comment by meitar** [comment]
@n8225 Commit 1b21ced72e0ff922dd60ad94d8f89e221fc66718 addresses your requested changes, I believe.

**Comment by nodiscc** [comment]
@meitar Everything looks good now, can you rebase to get rid of the merge commits in the MR? Thanks again

**Comment by meitar** [comment]
> can you rebase to get rid of the merge commits in the MR?

I do not know how to do this efficiently. Can you help?

The problem is that due to the long-ago fork, many if not all of the intermediary commits cannot be  cleanly merged with a `git rebase`. Unless there is some `git` command line technique that I am not aware of, you are asking me to re-do all the work I have done since I forked the project in January 2018. This is simply too much work for me to do manually again considering the alternative is a clean `git merge` command, which succeeds cleanly.

So, will you be able either to:

1. Show me how to cleanly rebase in such a way that does not cause so many conflicts during the rebasing process?
2. Merge the fork back in and ignore the divergence evident by the merge commits?

Thanks.

File: README.md [hunk 13]
```diff
@@ -851,16 +869,22 @@ _Games, game servers and control panels._
...
 - [TileServer GL](http://tileserver.readthedocs.io/) - Vector and raster maps with GL styles. Server side rendering by Mapbox GL Native. Map tile server for Mapbox GL JS, Android, iOS, Leaflet, OpenLayers, GIS via WMTS, etc. ([Source Code](https://github.com/klokantech/tileserver-gl)) `BSD-2-Clause` `Nodejs`
 - [TileServer PHP](https://github.com/klokantech/tileserver-php) - Serve map tiles from any PHP hosting. `BSD-2-Clause` `PHP`
 - [Traccar](https://www.traccar.org/) - Java application to track GPS positions. Supports loads of tracking devices and protocols, has an Android and iOS App. Has a web interface to view your trips. ([Demo](http://demo.traccar.org/), [Source Code](https://github.com/traccar)) `Apache-2.0` `Java`
+- [μlogger](https://github.com/bfabiszewski/ulogger-server) - Collect geolocation from users in real-time and display their GPS tracks on a website. ([Demo](http://ulogger.fabiszewski.net/)) `GPL-3.0` `PHP`
 - [uMap](https://umap.openstreetmap.fr/en/) - Create maps with OpenStreetMap layers in a minute and embed them in your site. ([Source Code](https://github.com/umap-project/umap)) `WTFPL` `Python`
```
Comment (n8225) [inline README.md]: Remove one of the uMap entries.

Comment (meitar) [reply README.md]: @n8225 Commit 1c009894ac472188e3331c57627e75f8dbce068f removes the duplication in uMap entries.

**Comment by nodiscc** [comment]
> how to cleanly rebase in such a way that does not cause so many conflicts

You can get a clean copy of Kickball/awesome-selfhosted, and replace README.md with your `remerge` version, then commit. The cleanest/easiest/git-like way is probably something like this (doesn't use `rebase`):

```bash
# clone the kickball/origin repo
~ $ git clone https://github.com/kickball/awesome-selfhosted && cd awesome-selfhosted

# add your fork as remote
~/awesome-selfhosted (master=)$ git remote add meitar https://github.com/meitar/awesome-selfhosted && git remote update

# create a clean branch for your PR, based on master
~/awesome-selfhosted (master=)$ git checkout -b remerge2

# get all updated files from the meitar/remerge branch
~/awesome-selfhosted (remerge2)$ git checkout meitar/remerge *
```

at this point your changes from origin/master...meitar/remerge are ready to commit. You can see this with `git status`. You can see what exact changes will be committed with `git diff --cached` (exactly the same diff as https://github.com/Kickball/awesome-selfhosted/pull/1797/files). Prepare a commit message in a text editor on the side,  you can use this one, I just checked:

re-integration of forked copy that contains many additional projects:
- Wording, typos, clarification, remove 'A', 'The'... at beginning of descriptions
- Add links to Source Code, Demo, and Clients lists for several projects
- Rename 'Gateways' section to 'Gateways and terminal sharing'
- Add Socioboard, LeapChat, LibreNews, Listmonk, InspIRCd, Patchwork, asciinema, GoTTY, tmate, OSRM, OpenTripPlanner, μlogger, Black Candy, CyberChef, oTranscribe, Revive Adserver, Open Event Server, sish
- Add 'Institutional repository and digital library software' subsection with DSpace, EPrints, Fedora Commons Repository, Islandora, Samvera Hyrax
- Add corresponding licenses

```bash
# when ready, commit, and paste the message:
~/awesome-selfhosted (remerge2 +)$ git commit

# push to your remote
~/awesome-selfhosted (remerge2 >)$ git push -u meitar/remerge2

```

Then submit a Pull Request from your `remerge2` branch to kickball/master: https://github.com/Kickball/awesome-selfhosted/compare/master...meitar:remerge2?expand=1

That's it. It will contain a single commit with an up-to date/concise message. There are other ways to do it (`git rebase`, `git add --patch`...) but this one is clean and easy.

clean

**Comment by meitar** [comment]
> It will contain a single commit with an up-to date/concise message.

@nodiscc Unless I am misunderstanding something, this also discards the development history along with authorship information for the committers who have contributed to my fork. That's very lossy (not what I would describe as "clean").

I find it hard to believe there is not another way? Admittedly, I am not aware of another way to do this, but I am also not confident removing the merge commits is worth discarding other people's efforts and taking credit for it myself. I am not really comfortable with discarding several other people's work just to make a commit history look unnecessarily linear.

**Comment by nodiscc** [comment]
Ah you're right, authorship info is lost. Apparently you can add multiple authors for a commit by adding to the message (there are only 2 authors in https://github.com/Kickball/awesome-selfhosted/pull/1797/commits, seems like some info is already lost):

```
Co-authored-by: name <name@example.com>
Co-authored-by: another-name <another-name@example.com>"
```

But right, let's just merge it.

**Comment by meitar** [comment]
Hmm, there *may* only actually be two authors but it's been so long that I don't recall. I know more people than that interacted with my fork but they may not have submitted actual patches. Else, this loss may be my own fault, prior to opening this pull request. In any event, I appreciate the care and approval for a simple merge.

**Pull request merged**